analysis/annotations/annotations.go

```diff
@@ -0,0 +1,378 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"fmt"
+	"go/ast"
+	"regexp"
+	"strings"
+
+	"github.com/awslabs/ar-go-tools/analysis/config"
+	"github.com/awslabs/ar-go-tools/internal/funcutil"
+	"golang.org/x/exp/slices"
+	"golang.org/x/tools/go/packages"
+	"golang.org/x/tools/go/ssa"
+)
+
+const annotationPrefix = "//argot:"
+
+// AnnotationKind characterizes the kind of annotations that can be used in the program.
+type AnnotationKind = int
+
+const (
+	// Sink is the kind of Sink(...) annotations.
+	Sink AnnotationKind = iota
+	// Source is the kind of Source(...) annotations
+	Source AnnotationKind = iota
+	// Sanitizer is the kind of Sanitizer(...) annotations
+	Sanitizer AnnotationKind = iota
+	// SetOptions is the kind of SetOptions(...) annotations
+	SetOptions AnnotationKind = iota
+)
+
+// AnyTag is the special tag used to match any other tag
+const AnyTag = "_"
+
+// TargetSpecifier is the type of target specifiers in the annotations: an annotation is of the shape
+// "//argot:<target specifier> <target specifier arguments>* <annotations>
+type TargetSpecifier = string
```

Comment:
Probably don't need a type alias here since `TargetSpecifier` should be a distinct type from `string`

Reply:
The goal here is just readablity/documentation (and refactoring ease later if we need to make this a more complex type).

Reply:
Agreed but that's possible with the syntax `type TargetSpecifier string` which I think is what you meant? Anyways this is more of a nit and a type alias mostly does the same thing